### PR TITLE
Benefits history page

### LINKED ISSUE #2829: BenefitsHistory component
And corresponding questions

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #2829.

Adds the benefits history page. There are some questions about when the date range is required, so I only added validation to check that the end date is after the start date.

### CODE REVIEW

**Comment by jbalboni** [comment]
screen shot 2016-09-02 at 8 39 32 am

**Comment by jbalboni** [comment]
@alexose Can you take a look? Don't merge this if you get through with it, it depends on another PR that's in progress.

File: src/js/edu-benefits/components/BenefitsHistoryFields.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import ErrorableRadioButtons from '../../common/components/form-elements/ErrorableRadioButtons';
+import ErrorableCheckbox from '../../common/components/form-elements/ErrorableCheckbox';
+import DateInput from '../../common/components/form-elements/DateInput';
+
+import { validateIfDirty, validateIfDirtyDateObj, isNotBlank, isValidDateField, isValidRepayingPeriodToDate } from '../utils/validations';
+import { yesNo } from '../utils/options-for-select';
+
+export default class BenefitsHistoryFields extends React.Component {
+  render() {
+    const isToDateValid = isValidRepayingPeriodToDate(this.props.data.activeDutyRepayingPeriod.fromDate, this.props.data.activeDutyRepayingPeriod.toDate);
```
Comment (alexose) [inline src/js/edu-benefits/components/BenefitsHistoryFields.jsx]: I'd break this up into a few lines for clarity

File: src/js/edu-benefits/utils/validations.js [hunk 2]
```diff
@@ -212,6 +212,17 @@ function isValidSeparatedDateField(date, dateEntered) {
   return true;
 }
 
+function isValidRepayingPeriodToDate(date, dateStarted) {
```
Comment (alexose) [inline src/js/edu-benefits/utils/validations.js]: The name seems overly specific given that it's just checking to see which date comes first.  Maybe something like `isDateBefore`?

Also, I would encourage the use of momentjs here (and everywhere we have to manipulate dates).

**Comment by alexose** [comment]
I have a few comments, but overall looking great

File: src/js/edu-benefits/utils/validations.js (hunk 2 above)
Comment (jbalboni) [inline src/js/edu-benefits/utils/validations.js]: I was mostly copying what had been done before, but that makes sense to me.

I was playing around with doing this in moment, but wasn't sure if there was some reason we avoided it for these validations in hca.

**Comment by jbalboni** [comment]
@alexose Updated based on your comments.

**Comment by alexose** [comment]
👍 

Will merge after CI

**Pull request merged**